gridstatus/tests/test_isone.py

```diff
@@ -122,21 +122,24 @@ def test_get_lmp_dst_boundaries(self, date, market):
         )
 
     def test_get_lmp_real_time_no_intervals_gets_current_data(self):
-        # Make sure when there are no intervals to fetch, the method still
-        # fetches the current data.
-        date = self.local_now().normalize() + pd.DateOffset(hours=2)
+        date = self.local_now() - pd.DateOffset(hours=2)
         end = date + pd.DateOffset(hours=1)
 
-        df = self.iso.get_lmp(
-            (date, end),
-            market=Markets.REAL_TIME_5_MIN,
-            verbose=VERBOSE,
-        )
+        # Mock the method _select_intervals_for_data_request in the class to return []
```

Comment:
I verified that the updated version of this test fails on main and passes with the changes in this PR.